MaxText/inference_microbenchmark.py

```diff
@@ -154,36 +148,45 @@ def write_results(results, filename=""):
 
 
 def print_results_for_analyze(results):
-  prefill_bucket_size_to_ms = {}
-  for k, v in results["Prefill"].items():
-    prefill_bucket_size_to_ms[int(k)] = round(v["prefill_time_in_ms"], 3)
-  print("\nFor usage in analyze_sharegpt.py :")
-  print(f"PREFILL_BUCKET_SIZE_TO_MS = {prefill_bucket_size_to_ms}")
-  print(f"SYSTEM_TIME_PER_DECODE_TOKEN_MS = {results['AutoRegressive']['ar_step_in_ms_per_seq']}")
+  if "Prefill" in results:
+    prefill_bucket_size_to_ms = {}
+    for k, v in results["Prefill"].items():
+      prefill_bucket_size_to_ms[int(k)] = round(v["prefill_time_in_ms"], 3)
+    print("\nFor usage in analyze_sharegpt.py :")
+    print(f"PREFILL_BUCKET_SIZE_TO_MS = {prefill_bucket_size_to_ms}")
+
+  if "AutoRegressive" in results:
+    print(f"SYSTEM_TIME_PER_DECODE_TOKEN_MS = {results['AutoRegressive']['ar_step_in_ms_per_seq']}")
 
 
 def main(config):
   engine = maxengine.MaxEngine(config)
   params = engine.load_params()
-  prefill_lengths = [64, 128, 256, 512, 1024]
-  benchmark_loop_iters = 10
+  prefill_lengths = [int(l) for l in config.inference_microbenchmark_prefill_lengths.split(",")]
```

Comment:
Does this work if you pass in a command line param like `--inference-microbenchmark-prefill-lengths="512,1024"` or something similar?

Reply:
Yeah, example commands:
to run a single prefill length: 
```
  inference_microbenchmark_prefill_lengths=1024
```
to run a single stage: 
```
  inference_microbenchmark_stages=generate
```